File: Factory/VehicleFactory.cs
// Interfaces y clases que implementan el patrón Factory
public interface IVehicle {
    void Drive();
}

public class Car : IVehicle {
    public void Drive() {
        Console.WriteLine("Driving a car.");
    }
}

public class Bike : IVehicle {
    public void Drive() {
        Console.WriteLine("Riding a bike.");
    }
}

// Clase Factory para crear objetos IVehicle
public class VehicleFactory {
    public static IVehicle GetVehicle(string type) {
        if (type == "Car") return new Car();
        if (type == "Bike") return new Bike();
        return null;
    }
}

// Uso del patrón Factory
class Program {
    static void Main(string[] args) {
        IVehicle vehicle = VehicleFactory.GetVehicle("Car");
        vehicle.Drive();
    }
}

File: Factory/vehicle_factory.py
# Clases que representan productos (Car y Bike)
class Car:
    def drive(self):
        print("Driving a car.")

class Bike:
    def drive(self):
        print("Riding a bike.")

# Clase Factory para crear objetos
class VehicleFactory:
    @staticmethod
    def get_vehicle(vehicle_type):
        if vehicle_type == "Car":
            return Car()
        elif vehicle_type == "Bike":
            return Bike()

# Uso del patrón Factory
vehicle = VehicleFactory.get_vehicle("Car")
vehicle.drive()
